### PR TITLE
Use go.uber.org/atomic in cmd/system-probe

### PR DESCRIPTION
**Pull request by djmitche**
### What does this PR do?

Uses go.uber.com/atomic to ensure alignment of atomic access in cmd/system-probe.

### Motivation

Alignment issues with atomic access.  See #11716.  The uses of atomic were very simple in these modules, and did not have bugs -- but the intent is to eliminate use of `sync/atomic` so that the module's bugs don't tempt us.

### Describe how to test/QA your changes

Any basic system-probe QA should detect issues here.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/system-probe/modules/oom_kill_probe.go [hunk 1]
```diff
@@ -36,16 +36,18 @@ var OOMKillProbe = module.Factory{
 	},
 }
 
-var _ module.Module = &oomKillModule{}
+var _ module.Module = &oomKillModule{
+	lastCheck: atomic.NewInt64(0),
```
Comment (brycekahle) [inline cmd/system-probe/modules/oom_kill_probe.go:40]: This variable is unused, and is only for type checks. I think you want this initialization on line 35 instead.

File: cmd/system-probe/modules/process.go [hunk 2]
```diff
@@ -41,26 +41,28 @@ var Process = module.Factory{
 	},
 }
 
-var _ module.Module = &process{}
+var _ module.Module = &process{
+	lastCheck: atomic.NewInt64(0),
```
Comment (brycekahle) [inline cmd/system-probe/modules/process.go:45]: Move to line 40

File: cmd/system-probe/modules/tcp_queue_tracer.go [hunk 3]
```diff
@@ -35,16 +35,18 @@ var TCPQueueLength = module.Factory{
 	},
 }
 
-var _ module.Module = &tcpQueueLengthModule{}
+var _ module.Module = &tcpQueueLengthModule{
+	lastCheck: atomic.NewInt64(0),
```
Comment (brycekahle) [inline cmd/system-probe/modules/tcp_queue_tracer.go:39]: Move to line 34

**Review by brycekahle — CHANGES_REQUESTED** [review]
Minor changes needed to TCPQ Length, Process, and OOM Kill checks

**Comment by djmitche** [comment]
face -> palm

That will teach me to make PRs during meetings

(..honestly, it probably won't)

**Review by brycekahle — APPROVED** [review]

**Comment by djmitche** [comment]
(unit test failures fixed in https://github.com/DataDog/datadog-agent/pull/12438)

**Comment by djmitche** [comment]
(rebased for lint fixes)

**Comment by sgnn7** [comment]
Merging since @djmitche will be returning after the freeze

**Pull request merged**